src/app/lorax/directives/share.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @fileOverview Share directive
+ * @author <a href="mailto:chris@work.co">Chris James</a>
+ * @author <a href="mailto:sneethling@mozilla.com">Schalk Neethling</a>
+ */
+define(['angular', 'jquery'], function (angular, $) {
+    'use strict';
+
+    /**
+     * Share directive
+     */
+    var ShareDirective = function () {
+        return {
+            restrict: 'A',
+            scope: true,
+            controller: ShareController
+        };
+    };
+
+    /**
+    * Controller for Share directive
+    * @constructor
+    */
+    var ShareController = function (
+        $scope,
+        dataService,
+        routesService
+    ) {
+
+        console.log('ShareController');
+
+        this._$scope = $scope;
+        this._dataService = dataService;
+        this._routesService = routesService;
+
+        // listen for $broadcast of 'share' which will be broadcast from:
+        // /src/app/lorax/controller/core.js
+        $scope.$on('share', this.share.bind(this));
+
+        this._dataService.getMain().then(function (model) {
+            this._$scope.content = {
+                share: model.getServices().share
+            };
+        }.bind(this));
+    };
+
+    /**
+    * Array of dependencies to be injected into controller
+    * @type {Array}
+    */
+    ShareController.$inject = [
```

Comment:
Is this even allowed anymore? I know you can't do app controllers this way in angular 1.3